src/tool_operate.c

```diff
@@ -1057,6 +1057,58 @@ static CURLcode operate_do(struct GlobalConfig *global,
           my_setopt_str(curl, CURLOPT_PINNEDPUBLICKEY, config->pinnedpubkey);
 
         if(curlinfo->features & CURL_VERSION_SSL) {
+          /* Verify if config->cert is a PKCS#11 URI and set the engine
+           * and config->cert_type if necessary */
+          if(config->cert) {
+            if(!strncmp(config->cert, "pkcs11:", 7)) {
+              if (!config->cert_type) {
+                config->cert_type = strdup("ENG");
+              }
+              if (!config->engine) {
+                config->engine = strdup("pkcs11");
```

Comment:
After some testing, I found out that setting the engine this way would break curl when a different crypto library is used (like GnuTLS). I added a commit to set this in the OpenSSL specific part of the library instead. As a consequence, other applications using libcurl will have the support for PKCS#11 URI when using OpenSSL.